waspc/src/Wasp/Generator/PackageJsonGenerator.hs

```diff
@@ -26,39 +80,22 @@ resolveNpmDeps ::
     [(D.Dependency, NpmDependenciesConflictError)]
     ([D.Dependency], [D.Dependency])
 resolveNpmDeps waspDeps userDeps =
-  if null conflictingUserDeps
-    then Right (waspDeps, userDepsNotInWaspDeps)
-    else Left conflictingUserDeps
+  case resolveDependencies waspDeps userDeps of
+    Left conflicts -> Left $ map convertConflict conflicts
+    Right (a, b) -> Right (a, b)
```

Comment:
You can shorten this as:
```hs
resolveNpmDeps waspDeps userDeps = left (map convertConflict) $ resolveDependencies waspDeps userDeps
```

What this does is map `map convertConflict` over `Left`.

Reply:
Cool, I didn't know about that. I needed `Control.Arrow` to make it work.